core/src/SYCL/Kokkos_SYCL.cpp

```diff
@@ -136,8 +132,15 @@ void SYCL::impl_static_fence(const std::string& name) {
       [&]() {
         // guard accessing all_queues
         std::lock_guard<std::mutex> lock(Impl::SYCLInternal::mutex);
-        for (auto& queue : Impl::SYCLInternal::all_queues)
-          Impl::SYCLInternal::fence(**queue);
+        for (auto& queue : Impl::SYCLInternal::all_queues) {
+          try {
+            (*queue)->wait_and_throw();
+          } catch (sycl::exception const& e) {
+            Kokkos::Impl::throw_runtime_exception(
+                std::string("There was a synchronous SYCL error:\n") +=
+                e.what());
+          }
```

Comment:
Why are you not using ` SYCLInternal::fence_helper` here?

Reply:
https://github.com/kokkos/kokkos/pull/4258#issue-718883570 says
> [...]  Also, the static fence can't really use the lowest level function anymore since we don't store the instance id with the queues. I decided to just copy the fence implementation for this case duplicating a few lines of code. Note that this case was treated in the Profiling interface separately already.